godot-macros/src/class/derive_godot_class.rs

```diff
@@ -273,17 +287,14 @@ fn parse_fields(class: &Struct) -> ParseResult<Fields> {
         let mut is_base = false;
         let mut field = Field::new(&named_field);
 
-        // #[base]
-        if let Some(parser) = KvParser::parse(&named_field.attributes, "base")? {
-            if let Some(prev_base) = base_field.as_ref() {
-                return bail!(
-                    parser.span(),
-                    "#[base] allowed for at most 1 field, already applied to `{}`",
-                    prev_base.name
-                );
-            }
+        // Base<T> type inference
+        if path_ends_with_complex(&field.ty, "Base") {
             is_base = true;
-            parser.finish()?;
+        }
+
+        // deprecated #[base]
+        if KvParser::parse(&named_field.attributes, "base")?.is_some() {
+            has_deprecated_base = true;
         }
```

Comment:
If we're going to have an explicit deprecation warning for this, why not make it work like it used to at least until we remove the deprecation warning? As is this deprecation warning doesn't help that much if your `#[base]` declaration is incompatible with the new way of finding base fields (like say with a type alias).

Reply:
In practice, most existing code will keep working due to it using `Base<T>` already. But it should not be too hard to make `#[base]` have the same semantics as `#[hint(base)]`, maybe I can do that.

Reply:
Updated, also quickly tested it by replacing one of the `#[hint(base)]` with `#[base]` in integration tests.